src/test/java/org/cryptomator/fusecloudaccess/CloudAccessFSTest.java

```diff
@@ -44,26 +61,336 @@ public void testInterrupt() throws InterruptedException {
 		Assertions.assertEquals(-ErrorCodes.EINTR(), actualResult.get());
 	}
 
-	@DisplayName("testReturnOrTimeoutOnExecutionException")
+	@DisplayName("test returnOrTimeout() returns EIO on ExecutionException")
 	@Test
 	public void testExecution() {
-		CompletableFuture future = CompletableFuture.completedFuture(0)
-				.thenApply(i -> {
-					throw new TestException();
-				});
+		CompletableFuture future = CompletableFuture.failedFuture(new Exception());
 		Assertions.assertEquals(-ErrorCodes.EIO(), cloudFs.returnOrTimeout(future));
 	}
 
-	@DisplayName("testReturnOrTimeoutOnTimeout")
+	@DisplayName("test returnOrTimeout() return ETIMEDOUT on timeout")
 	@Test
 	public void testTimeout() {
 		Assertions.assertEquals(-ErrorCodes.ETIMEDOUT(), cloudFs.returnOrTimeout(new CompletableFuture<>()));
 	}
 
-	private static class TestException extends RuntimeException {
-		TestException() {
-			super();
-			setStackTrace(new StackTraceElement[] {});
+	static class GetAttrTests {
+
+		private static final Runtime RUNTIME = Runtime.getSystemRuntime();
+
+		private CloudProvider provider;
+		private CloudAccessFS cloudFs;
+		private FileStat fileStat;
+
+		@BeforeEach
+		public void setup() {
+			provider = Mockito.mock(CloudProvider.class);
+			cloudFs = new CloudAccessFS(provider, CloudAccessFSTest.TIMEOUT);
+			fileStat = new FileStat(RUNTIME);
+		}
+
+		@DisplayName("getAttr() returns 0 on success")
+		@Test
+		public void testGetAttrSuccess() {
+			Mockito.when(provider.itemMetadata(PATH))
+					.thenReturn(CompletableFuture.completedFuture(CloudItemMetadataProvider.ofPath(PATH)));
+
+			Assertions.assertEquals(0, cloudFs.getattr(PATH.toString(), fileStat));
+		}
+
+		@DisplayName("getAttr() returns ENOENT when resource is not found.")
+		@Test
+		public void testGetAttrNotFound() {
+			Mockito.when(provider.itemMetadata(PATH))
+					.thenReturn(CompletableFuture.failedFuture(new NotFoundException()));
+
+			Assertions.assertEquals(-ErrorCodes.ENOENT(), cloudFs.getattr(PATH.toString(), fileStat));
+		}
+
+		@DisplayName("getAttr() returns EIO on CloudProviderException.")
+		@Test
+		public void testGetAttrCloudError() {
+			Mockito.when(provider.itemMetadata(PATH))
+					.thenReturn(CompletableFuture.failedFuture(new CloudProviderException()));
+
+			Assertions.assertEquals(-ErrorCodes.EIO(), cloudFs.getattr(PATH.toString(), fileStat));
+		}
+
+		@DisplayName("getAttr() returns EIO on any exception.")
+		@Test
+		public void testGetAttrSomeException() {
+			Mockito.when(provider.itemMetadata(PATH))
+					.thenReturn(CompletableFuture.failedFuture(new Exception()));
+
+			Assertions.assertEquals(-ErrorCodes.EIO(), cloudFs.getattr(PATH.toString(), fileStat));
+		}
```

Comment:
Can these two tests be collapsed to a ParameterizedTest with all the exception classes for which EIO is expected?

Reply:
resolved by da080597d67cce68cd7f985d6edbd924bb785bee